### PR TITLE
Add a release cadence page.

### PR DESCRIPTION
**Pull request by geeknoid**
Staging site: https://geeknoid.github.io/istio.github.io/about/release-cadence.html

### CODE REVIEW

File: _about/release-cadence.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+title: Build & Release Cadence
+description: How we manage, number, and support Istio releases.
+weight: 5
+---
+
+We produce new builds of Istio on a daily basis.
+Once a week, one of these daily build undergoes additional
+qualification and gets anointed the weekly release. Once a month, one of the weekly releases undergo
+additional qualifications and gets anointed as the monthly release. Once a quarter or so, one of the monthly
+releases undergoes additional qualifications and gets anointed as a long term supported (LTS) release.
```
Comment (linsun) [inline _about/release-cadence.md]: I'm not familiar with the additional qualification for weekly release.  Is this doc-ed anywhere?

File: _about/release-cadence.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Each release type implies different levels of quality and support:
+
+|Release Type | Support Level | Typical Use
+|-------------+---------------+------------
+|Daily   | No support           | Dangerous, may not be fully reliable. Useful to experiment with.
+|Weekly  | No support           | Dangerous, may not be fully reliable. Useful to experiment with.
+|Monthly | Support is provided for the current monthly release only | Safe to deploy in production. Usually only adopted by bleeding edge users, or users seeking specific features.
```
Comment (linsun) [inline _about/release-cadence.md]: Do we really want to say monthly is safe to deploy in prod?

Comment (geeknoid) [reply _about/release-cadence.md]: That came from this doc: https://docs.google.com/document/d/1scC_f7nObwBut_xxE6lUfvcHBybmttUsJDMwXD3gcO4/edit#heading=h.ob68p14f45rs

Let's talk about it during the TOC meeting.

File: _about/release-cadence.md (hunk 1 above)
Comment (geeknoid) [reply _about/release-cadence.md]: That came from https://github.com/istio/istio/blob/master/release/README.md and from https://docs.google.com/document/d/1_SNEDOSGyw69UnLC9kYY7bTfxVszTE-W-mLdtPVJuOA/edit

Comment (hklai) [reply _about/release-cadence.md]: Weekly release has not been implemented.
And I will send a PR to update the ETA in https://github.com/istio/istio/blob/master/release/README.md

File: _about/release-cadence.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+|LTS     | Support is provided until 3 months after the next LTS                | Safe to deploy in production. Users are encouraged to upgrade to these releases as soon as possible.
+
+In this context, *support* means that we will produce patch releases for critical issues and offer technical assistance.
+In addition, longer-term support may be available from 3rd parties and partners.
+
+Prior to Istio 0.8, we increased the product's version number for every monthly release. Effective with 0.8,
+we will increase the product's version number only for LTS releases.
+
+You can find all available releases on our [releases page](https://github.com/istio/istio/releases).
```
Comment (hklai) [inline _about/release-cadence.md]: daily releases can be found in https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/, not on the github release page.

File: _about/release-cadence.md (hunk 1 above)
Comment (hklai) [reply _about/release-cadence.md]: Weekly release pipeline has not been implemented.

I will send a PR to update the ETA in https://github.com/istio/istio/blob/master/release/README.md.

File: _about/release-cadence.md (hunk 3 above)
Comment (ldemailly) [reply _about/release-cadence.md]: or with 1 level of indirection so this page doesn't need to be updated when something changes, I would link to https://github.com/istio/istio/wiki/Daily-builds instead which has additional information

**Comment by geeknoid** [comment]
Updated following the TOC meeting. PTAL.

Thanks.

File: _about/release-cadence.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+title: Build & Release Cadence
+description: How we manage, number, and support Istio releases.
+weight: 5
+---
+
+We produce new builds of Istio on a daily basis. Around once a month or so, we take one of these daily
+builds and run it through a number of additional qualification tests and tag the build as an Alpha release.
+Around once a quarter or so, we take one of these Alpha releases, run through a bunch more tests
+and tag the build as a Long Term Support (LTS) release. Finally, if we find something wrong with an Alpha
+or LTS release, we issue patches.
```
Comment (ldemailly) [inline _about/release-cadence.md]: historically we have not patched the monthly release after the first qualified one. not sure if we want to commit to fixing critical issues in monthlies vs ask people to wait for the next one

File: _about/release-cadence.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+Prior to Istio 0.8, we increased the product's version number on a monthly basis. Effective with 0.8,
+we will increase the product's version number only for LTS releases.
+
+Our naming scheme for LTS releases is:
+
+```plain
+<major>.<minor>.<patch level>
+```
+
+where `minor` is increased for every LTS release, and `patch level` counts the number of patches for the
```
Comment (ldemailly) [inline _about/release-cadence.md]: numbe of patch releases

(there will be more than 1 bug fix per patch release)

File: _about/release-cadence.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+current LTS release.
+
+For Alpha releases, our numbering scheme is:
+
+```plain
+<major>.<minor>.<patch level>-alpha<alpha count>
+```
+
+where `minor` represents the number of the next LTS release, `patch level` counts the number of
+patches for the Alpha release, and `alpha count` increases by one for every Alpha release leading up
```
Comment (ldemailly) [inline _about/release-cadence.md]: that doesn’t follow sem ver
(or forces to make the first lts not be .0 )

File: _about/release-cadence.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+or LTS release, we issue patches.
+
+The different types (Daily, Alpha, LTS) represent different product quality levels and different levels of support
+from the Istio team. In this context, *support* means that we will produce patch releases for critical issues and
+offer technical assistance. Separately, 3rd parties and partners may offer longer-term support solutions.
+
+|Type          | Support Level                                         | Quality and Recommended Use
+|--------------+-------------------------------------------------------+----------------------------
+|Daily Build   | No support                                            | Dangerous, may not be fully reliable. Useful to experiment with.
+|Alpha Release | Support is provided for the latest Alpha release only | Expected to be quite stable, but use in production should be limited to an as-needed basis. Usually only adopted by bleeding edge users, or users seeking specific features.
```
Comment (hklai) [inline _about/release-cadence.md]: Are we going to produce release notes for alpha releases going forward?

File: _about/release-cadence.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+where `minor` is increased for every LTS release, and `patch level` counts the number of patches for the
+current LTS release.
+
+For Alpha releases, our numbering scheme is:
+
+```plain
+<major>.<minor>.<patch level>-alpha<alpha count>
+```
+
+where `minor` represents the number of the next LTS release, `patch level` counts the number of
```
[the remark this answers is not in the document]
Comment (ldemailly) [reply _about/release-cadence.md]: and patch (again you can't have 0.9.3-alpha2 be before 0.9.0

also is the first one 0.9.0-alpha0 or alpha1 ?

what is 0.9.0-alpha2 vs 0.9.2-alpha1 for instance ?

File: _about/release-cadence.md (hunk 7 above)
Comment (geeknoid) [reply _about/release-cadence.md]: I don't think so, only full notes for LTS releases

File: _about/release-cadence.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (geeknoid) [reply _about/release-cadence.md]: How does this model conflict with what's defined in https://semver.org/?

First Alpha release for 0.9 would be 0.9.0-alpha.0

0.9.0-alpha.2 means "the third alpha release for 0.9.0"
0.9.2-alpha.1 means "the second alpha release for 0.9.2"

So as per [this](https://semver.org/#spec-item-9), the stuff to the right of the - indicates these are pre-release versions.

[the remark this answers is not in the document]
Comment (hklai) [reply _about/release-cadence.md]: Say we release the following over the course of few months. They are all cut at head in chronological order.

0.9.0-alpha.0, 0.9.0-alpha.1, 0.9.0-alpha.2, and 0.9.0.

When we patch 0.9.0, we create 0.9.1. *If* we apply the same patch to the alpha releases, the patched 0.9.0-alpha.2 will become 0.9.1-alpha.2, right? Strictly speaking, 0.9.1-alpha.2 is not really the third alpha release of 0.9.1.

I think this scheme works best if we only patch LTS releases but not alpha releases, so we won't end up in a confusing situation like above.

**Comment by geeknoid** [comment]
OK, that's right. Patching of alpha releases is a different beast than
patching of LTS releases. We can't overload the number like that. So
following semver, we would need to add another number:

<major>.<minor>.<patch level>-alpha.<alpha count>.<alpha patch level>

File: _about/release-cadence.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
+---
+title: Build & Release Cadence
+description: How we manage, number, and support Istio releases.
+weight: 5
+---
+
+We produce new builds of Istio on a daily basis. Around once a month or so, we take one of these daily
+builds and run it through a number of additional qualification tests and tag the build as an Alpha release.
```
Comment (smawson) [inline _about/release-cadence.md]: I feel weird calling these alpha releases, since alpha has a specific meaning in terms of features, and this is unrelated to that. Can we just call them monthlies?  Prerelease? Anything other than alpha/beta/stable would be useful here.

File: _about/release-cadence.md (hunk 4 above)
Comment (smawson) [reply _about/release-cadence.md]: I'd almost say we don't bother mentioning patching of the monthlies. Over time we don't want to spend time doing that, those patches should just roll into a new monthly RC.

File: _about/release-cadence.md (hunk 7 above)
Comment (smawson) [reply _about/release-cadence.md]: We should at a minimum point at an automated set of changes between monthlies, but not do a curated set of notes.

File: _about/release-cadence.md (hunk 9 above)
Comment (geeknoid) [reply _about/release-cadence.md]: Oh, yeah, the use of alpha here will be confusing with what we use for individual features. I'll change to 'prerelease'.

Comment (ldemailly) [reply _about/release-cadence.md]: standard is "rc" for the short form in the semver suffix

File: _about/release-cadence.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+builds and run it through a number of additional qualification tests and tag the build as a Snapshot release.
+Around once a quarter or so, we take one of these Snapshot releases, run through a bunch more tests
+and tag the build as a Long Term Support (LTS) release. Finally, if we find something wrong with an
+LTS release, we issue patches.
+
+The different types (Daily, Snapshot, LTS) represent different product quality levels and different levels of support
+from the Istio team. In this context, *support* means that we will produce patch releases for critical issues and
+offer technical assistance. Separately, 3rd parties and partners may offer longer-term support solutions.
+
+|Type             | Support Level                                         | Quality and Recommended Use
```
Comment (ldemailly) [inline _about/release-cadence.md]: table doesn't render on github md's renderer: https://github.com/geeknoid/istio.github.io/blob/64a98d66f5ccb53f698ac72d9be86ac28cab3592/_about/release-cadence.md

File: _about/release-cadence.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+<major>.<minor>.<patch level>
+```
+
+where `minor` is increased for every LTS release, and `patch level` counts the number of patches for the
+current LTS release.
+
+For Snapshot releases, our numbering scheme is:
+
+```plain
+<major>.<minor>.<patch level>-snapshot.<snaoshot count>
```
Comment (ldemailly) [inline _about/release-cadence.md]: snaoshot-> snapshot

File: _about/release-cadence.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+```
+
+where `major` and `minor` represent the number of the next LTS release, `patch level` counts the number of
+patches for the Snapshot release, and `snapshot count` starts at 0 and increases for every Snapshot release leading up
+to the next LTS.
```
Comment (ldemailly) [inline _about/release-cadence.md]: this is confusing... what is patchlevel vs snapshot count - when do each change/seems backward

it seems to read like we would do 1.0.0-snapshot.0 followed by 1.0.1-snapshot.0 if making a patch, followed by 1.0.0-snapshot.1 for the next month... which is backward ?

Comment (ldemailly) [reply _about/release-cadence.md]: it should all be 1.0.0-rc1.0 then 1.0.0.rc1.1 then 1.0.0.rc2.0 etc... the first 3 digits are frozen to "next lts" by definition

**Comment by geeknoid** [comment]
The model follows quite closely with the semver approach,

* major.minor.patch represents the labeling for the next LTS release.

* The -snapshot-count suffix reflects monthly-ish releases we cut on the
way to an LTS.

* Snapshot patch level is about us issuing an (unlikely) patch to an
existing snapshot. More likely, we would probably not release a patch and
just say "wait til the next snapshot to get this fix". But you never know...

File: _about/release-cadence.md (hunk 10 above)
Comment (geeknoid) [reply _about/release-cadence.md]: See https://geeknoid.github.io/istio.github.io/about/release-cadence.html

File: _about/release-cadence.md [hunk 13]
```diff
@@ -33,24 +34,25 @@ we will increase the product's version number only for LTS releases.
...
 
 ```plain
-<major>.<minor>.<patch level>-snapshot.<snaoshot count>
+<major>.<minor>.<LTS patch level>-snapshot.<snapshot count>
 ```
 
-where `major` and `minor` represent the number of the next LTS release, `patch level` counts the number of
-patches for the Snapshot release, and `snapshot count` starts at 0 and increases for every Snapshot release leading up
-to the next LTS.
+where `major` and `minor` and `LTS patch level` represent the next LTS, and
```
Comment (ldemailly) [inline _about/release-cadence.md]: that's clearer thank you, but you might as well replace <LTS patch level> by 0 as before an LTS release happens the patch level will always be 0
(though we may do -rc for patch levels I suppose but those wouldn't be monthlies anyway)

I would recommend to replace "snapshot" with "rc" in the string for both familiarity and brievety

File: _about/release-cadence.md [hunk 14]
```diff
@@ -33,24 +34,25 @@ we will increase the product's version number only for LTS releases.
...
+`snapshot count` starts at 0 and increases for every Snapshot leading up to the
+next LTS.
 
-In the unlikely event we need to issue a patch to a snapshot release, it is numbered as:
+In the unlikely event we need to issue a patch to a snapshot, it is numbered as:
 
 ```plain
-<major>.<minor>.<patch level>-snapshot.<snaoshot count>.<snapshot patch level>
+<major>.<minor>.<LTS patch level>-snapshot.<snaoshot count>.<snapshot patch level>
```
Comment (ldemailly) [inline _about/release-cadence.md]: snaoshot -> snapshot here too (isn't the build supposed to find typos?)

File: _about/release-cadence.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+<major>.<minor>.<LTS patch level>
+```
+
+where `minor` is increased for every LTS release, and `LTS patch level` counts the number of patches for the
+current LTS release.
+
+For snapshot releases, our numbering scheme is:
+
+```plain
+<major>.<minor>.<LTS patch level>-snapshot.<snapshot count>
```
Comment (ldemailly) [inline _about/release-cadence.md]: can you replace _LTS patch level_ by 0 for clarity here and the below example ?

**Comment by ldemailly** [comment]
lgtm thx!

File: _about/release-cadence.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+Prior to Istio 0.8, we increased the product's version number on a monthly basis. Effective with 0.8,
+we will increase the product's version number only for LTS releases.
+
+Our naming scheme for LTS releases is:
+
+```plain
+<major>.<minor>.<LTS patch level>
+```
+
+where `minor` is increased for every LTS release, and `LTS patch level` counts the number of patches for the
```
Comment (hklai) [inline _about/release-cadence.md]: It may be a good idea to mention that a patch is always created on top of the corresponding LTS release, and not from head.

File: _about/release-cadence.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+current LTS release.
+
+For snapshot releases, our numbering scheme is:
+
+```plain
+<major>.<minor>.0-snapshot.<snapshot count>
+```
+
+where `major`, `minor` and `LTS patch level` represent the next LTS, and
```
Comment (hklai) [inline _about/release-cadence.md]: Remove `LTS patch level`

Comment (hklai) [reply _about/release-cadence.md]: Also, I wonder if we should use a different character between "snapshot" and <snapshot count>? Say '_' instead?

Comment (geeknoid) [reply _about/release-cadence.md]: The . is what the semver spec says to use...

**Comment by geeknoid** [comment]
All righty. hopefully final call for comments.

**Comment by smawson** [comment]
Just took a look over the staging version and looks great, approved from me!
